src/Bundle/Resources/config/services.xml

```diff
@@ -43,20 +43,45 @@
         <service id="sylius.grid.configuration_removals_handler" class="Sylius\Component\Grid\Configuration\GridConfigurationRemovalsHandler"/>
         <service id="Sylius\Component\Grid\Configuration\GridConfigurationRemovalsHandlerInterface" alias="sylius.grid.configuration_removals_handler" />
 
+        <service id="sylius.grid.configuration_sorting_handler" class="Sylius\Component\Grid\Configuration\GridConfigurationSortingHandler"/>
+        <service id="Sylius\Component\Grid\Configuration\GridConfigurationSortingHandlerInterface" alias="sylius.grid.configuration_sorting_handler" />
+
         <service id="sylius.grid.array_grid_provider" class="Sylius\Component\Grid\Provider\ArrayGridProvider">
-            <argument type="service" id="sylius.grid.array_to_definition_converter" />
             <argument>%sylius.grids_definitions%</argument>
-            <argument type="service" id="sylius.grid.configuration_extender" />
-            <argument type="service" id="sylius.grid.configuration_removals_handler" />
+            <argument type="service" id="sylius.grid.grid_factory" />
             <tag name="sylius.grid_provider" key="array" priority="-200"/>
         </service>
         <service id="Sylius\Component\Grid\Provider\ArrayGridProvider" alias="sylius.grid.array_grid_provider" />
 
-        <service id="sylius.grid.service_grid_provider" class="Sylius\Bundle\GridBundle\Provider\ServiceGridProvider">
+        <service id="sylius.grid.grid_factory" class="Sylius\Component\Grid\Factory\GridFactory">
```

Comment:
Maybe...
```suggestion
        <service id="sylius.grid.factory" class="Sylius\Component\Grid\Factory\GridFactory">
```

Reply:
Ideally, to be consistent with Sylius/Sylius, it would be:
```suggestion
        <service id="sylius_grid.factory.grid" class="Sylius\Component\Grid\Factory\GridFactory">
```
but it wouldn't be consistent here, so I would be for:
```suggestion
        <service id="sylius.grid.factory.grid" class="Sylius\Component\Grid\Factory\GridFactory">
```
🤔